components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -337,7 +337,7 @@ public String addAPI(API.APIBuilder apiBuilder) throws APIManagementException {
                 if (StringUtils.isEmpty(apiBuilder.getApiDefinition())) {
                     apiBuilder.apiDefinition(apiDefinitionFromSwagger20.generateSwaggerFromResources(apiBuilder));
                 }
-                if (apiBuilder.getPermission() != null && !("").equals(apiBuilder.getPermission())) {
+                if (apiBuilder.getPermission() != null && !StringUtils.isEmpty(apiBuilder.getPermission())) {
```

Comment:
string utils check both null case and empty case. We can remove the null check :+1: